pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+)
+
+const (
+	scorePlugin1 = "score-plugin-1"
+	scorePlugin2 = "score-plugin-2"
+	scorePlugin3 = "score-plugin-3"
+)
+
+var _ = NormalizeScorePlugin(&TestNormalizeScorePlugin1{})
+var _ = NormalizeScorePlugin(&TestNormalizeScorePlugin2{})
+
+type TestNormalizeScorePlugin1 struct {
+	// If fail is true, NormalizeScore will return error status.
+	fail bool
+}
+
+// NewNormalizeScorePlugin1 is the factory for Normalize Score plugin 1.
+func NewNormalizeScorePlugin1(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+	return &TestNormalizeScorePlugin1{}, nil
+}
+
+// NewNormalizeScorePlugin1InjectFailure creates a new TestNormalizeScorePlugin1 which will
+// return an error status for NormalizeScore.
+func NewNormalizeScorePlugin1InjectFailure(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+	return &TestNormalizeScorePlugin1{fail: true}, nil
+}
+
+func (pl *TestNormalizeScorePlugin1) Name() string {
+	return scorePlugin1
+}
+
+func (pl *TestNormalizeScorePlugin1) NormalizeScore(pc *PluginContext, scores NodeScoreList) *Status {
+	if pl.fail {
+		return NewStatus(Error, "injecting failure.")
+	}
+	// Simply decrease each node score by 1.
+	for i := range scores {
+		scores[i] = scores[i] - 1
+	}
+	return nil
+}
+
+type TestNormalizeScorePlugin2 struct{}
+
+// NewNormalizeScorePlugin2 is the factory for Normalize Score plugin 2.
+func NewNormalizeScorePlugin2(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+	return &TestNormalizeScorePlugin2{}, nil
+}
+
+func (pl *TestNormalizeScorePlugin2) Name() string {
+	return scorePlugin2
+}
+
+func (pl *TestNormalizeScorePlugin2) NormalizeScore(pc *PluginContext, scores NodeScoreList) *Status {
+	// Simply force each node score to 5.
+	for i := range scores {
+		scores[i] = 5
+	}
+	return nil
+}
+
+func TestFramework_RunNormalizeScorePlugins(t *testing.T) {
```

Comment:
Do we prefer to use underscores as part of the name in test functions?

Reply:
> Looks good, but I think partially fixes would close that issue after this gets merged.

I think `fixes` is the key word to trigger the close? I can rephrase this to get rid of `fixes`

Reply:
> Do we prefer to use underscores as part of the name in test functions?

Quickly looked at some other tests and didn't see underscore used. I can remove it.